### PR TITLE
KT-34142 sortedMapOf overload with Comparator and items

### PR DESCRIPTION
**Pull request by valery1707**
https://youtrack.jetbrains.com/issue/KT-34142

### CODE REVIEW

File: libraries/stdlib/jvm/src/kotlin/collections/MapsJVM.kt [hunk 1]
```diff
@@ -76,6 +76,17 @@ public fun <K, V> Map<out K, V>.toSortedMap(comparator: Comparator<in K>): Sorte
...
 
+/**
+ * Returns a new [SortedMap] with the specified contents, given as a list of pairs
+ * where the first value is the key and the second is the value.
+ *
+ * The resulting [SortedMap] determines the equality and order of keys according to the sorting order provided by the given [comparator].
+ *
+ * @sample samples.collections.Maps.Instantiation.sortedMapWithComparatorFromPairs
+ */
+public fun <K, V> sortedMapOf(comparator: Comparator<in K>, vararg pairs: Pair<K, V>): SortedMap<K, V> =
```
Comment (ilya-g) [inline libraries/stdlib/jvm/src/kotlin/collections/MapsJVM.kt:87]: Needs to be annotated with `@SinceKotlin("1.4")`

File: libraries/stdlib/samples/test/samples/collections/maps.kt [hunk 2]
```diff
@@ -43,6 +43,13 @@ class Maps {
             assertPrints(map.values, "[2, 3, 1]")
         }
 
+        @Sample
+        fun sortedMapWithComparatorFromPairs() {
+            val map = sortedMapOf(compareBy<String> { it.length }.thenBy { it }, Pair("abc", 1), Pair("c", 3), Pair("bd", 4), Pair("bc", 2))
+            assertPrints(map.keys, "[c, bc, bd, abc]")
+            assertPrints(map.values, "[c, bc, bd, abc]")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/maps.kt:50]: `values` collection should contain integers

**Comment by ilya-g** [comment]
I've fixed the remaining issues and pushed c023a02884fd18cd43ccb73fba5a80551091b6e0 to master.
Thank you for the PR.

**Comment by valery1707** [comment]
Sorry for the long delay - I did not expect any activity after 8 months of silence - and skipped this conversation

**Pull request closed without merging**